Prepare release 3.0.0-beta.0

fionaliao opened the pull request:
Updated version numbers and CHANGELOG

juliusv commented on CHANGELOG.md:
```diff
@@ -1,18 +1,30 @@
-
-## unreleased
-
+* [CHANGE] UTF-8: Enable UTF-8 support by default. Prometheus now allows all UTF-8 characters in metric and label names. The corresponding `utf8-name` feature flag has been removed. #14770
```
It seems like #14770 is a different PR?

fionaliao replied:
Whoops, you're right. Fixed that and another PR ref in https://github.com/prometheus/prometheus/pull/14853/commits/1682d6192c128199bf1c96a333be466df0926dd7

juliusv commented on CHANGELOG.md:
```diff
@@ -1,18 +1,30 @@
-
-## unreleased
-
+* [CHANGE] UTF-8: Enable UTF-8 support by default. Prometheus now allows all UTF-8 characters in metric and label names. The corresponding `utf8-name` feature flag has been removed. #14705
```
For the new UI, you could add something like this:

```
* [CHANGE] UI: The old web UI has been replaced by a completely new one that is less cluttered and adds a few new features (PromLens-style tree view, better metrics explorer, "Explain" tab). However, it is still missing some features of the old UI (notably, exemplar display and heatmaps). To switch back to the old UI, you can use the feature flag `--enable-feature=old-ui` for the time being.
```

In case that's too long, feel free to redact. We also usually put some general release info at the top of the list when there's something major going on, so it probably should also be mentioned there in the end.

juliusv replied:
Oh and the PR that merged the new UI was #14872.

jan--f commented on web/ui/package.json:
```diff
@@ -28,5 +28,5 @@
     "typescript": "^4.9.5"
   },
-  "version": "0.54.1"
+  "version": "3.0.0-beta.0"
```
Not sure about this change here. IF we follow the same version pattern as for go modules here this should be `0.300.0-beta.0` I believe. Same for the other locations.

fionaliao replied:
Changed to `0.300.0-beta.0` in https://github.com/prometheus/prometheus/pull/14853/commits/962ebee4f9b324ede8d3690e7e834a420e616e15

Using the new `get_module_version.sh` script to update the version when running `make ui-bump-version`

juliusv replied:
Yeah, I think that should be fine for the UI as well. @Nexucis can confirm or deny, but I don't see anything different we should do there.

Nexucis replied:
yes that sounds right to me.

jan--f commented:
Another main sync is in https://github.com/prometheus/prometheus/pull/14879. I updated the changelog there too with changes that went into main since the last release.

fionaliao commented:
> Another main sync is in #14879. I updated the changelog there too with changes that went into main since the last release.

Rebased PR off sync from #14879. Note that there were some duplicate changelog entries between that sync and this PR, those have been merged.

fionaliao commented:
Changelog should be update to date (for now).

UI tests are failing though

juliusv commented:
UI test failure says `npm ERR! notarget No matching version found for @prometheus-io/codemirror-promql@0.300.0-beta.0.`

@Nexucis do you know what'd be the issue here? Something about the new versioning scheme maybe?

Nexucis commented:
since the old ui is not part of the workspace anymore, it cannot get the unrelease codemirror version.
I would advise to avoid updating codemirror in the old UI or to add it back to the workspace

juliusv commented:
As far as I can see, the `ui_release.sh` script in @fionaliao's branch is already my latest one that should only update the verisons for the workspace-ed modules, so the old React app should be excluded already: https://github.com/fionaliao/prometheus/blob/release-3.0.0-beta/scripts/ui_release.sh#L73C1-L89C2

But probably the old React app dependency versions were bumped in this PR already before my updated script made it in. So maybe the solution is as simple as pushing another commit here that removes the `@prometheus-io/codemirror-promql` version bump in `web/ui/react-app/package.json`.

Nexucis commented:
> So maybe the solution is as simple as pushing another commit here that removes the @prometheus-io/codemirror-promql version bump in web/ui/react-app/package.json.

yup just rollback to the old version of codemirror in the old UI and then re-run `npm install` to update the file `package-lock.json`

Nexucis commented:
since old ui is removed for the workspace, `npm install` need to be ran in the folder `react-app` too

fionaliao commented:
@juliusv @Nexucis thanks for your help, the issue was that I ran the ui_release.sh script before the sync from the 3.0 branch. The tests pass now.

@jan--f I agree, undrafting now. Waiting go tests check to finish (they failed the first time but didn't see a clear issue so rerunning)

Nexucis commented:
Tests are failing because of `no space left on device`. Re-running the job, perhaps there is an issue with a github action runner

```bash
2024-09-10T11:57:09.7240258Z FAIL	github.com/prometheus/prometheus/storage/remote/azuread [build failed]
2024-09-10T11:57:09.7241832Z ?   	github.com/prometheus/prometheus/storage/remote/googleiam	[no test files]
2024-09-10T11:57:09.7252974Z github.com/prometheus/prometheus/storage/remote/otlptranslator/prometheus: write /tmp/go-build1892333022/b1480/importcfg: no space left on device
2024-09-10T11:57:09.7254161Z FAIL	github.com/prometheus/prometheus/storage/remote/otlptranslator/prometheus [build failed]
2024-09-10T11:57:09.7265729Z github.com/prometheus/prometheus/storage/remote/otlptranslator/prometheusremotewrite: write /tmp/go-build1892333022/b1483/importcfg: no space left on device
2024-09-10T11:57:09.7267928Z FAIL	github.com/prometheus/prometheus/storage/remote/otlptranslator/prometheusremotewrite [build failed]
2024-09-10T11:57:09.7274105Z github.com/prometheus/prometheus/template: mkdir /tmp/go-build1892333022/b1486/: no space left on device
2024-09-10T11:57:09.7274937Z FAIL	github.com/prometheus/prometheus/template [build failed]
2024-09-10T11:57:09.7283714Z github.com/prometheus/prometheus/tracing: write /tmp/go-build1892333022/b1489/importcfg: no space left on device
2024-09-10T11:57:09.7284549Z FAIL	github.com/prometheus/prometheus/tracing [build failed]
2024-09-10T11:57:09.7304695Z github.com/prometheus/prometheus/tsdb: write /tmp/go-build1892333022/b1492/importcfg: no space left on device
2024-09-10T11:57:09.7305522Z FAIL	github.com/prometheus/prometheus/tsdb [build failed]
2024-09-10T11:57:10.0386033Z # github.com/prometheus/prometheus/tsdb/agent [github.com/prometheus/prometheus/tsdb/agent.test]
2024-09-10T11:57:10.0387454Z compile: writing output: write $WORK/b1495/_pkg_.a: no space left on device
2024-09-10T11:57:10.0388519Z FAIL	github.com/prometheus/prometheus/tsdb/agent [build failed]
```

juliusv commented:
> Tests are failing because of `no space left on device`. Re-running the job, perhaps there is an issue with a github action runner

@Nexucis I saw it fail again. I'm re-running it again, but generally I saw this same error before with my `mantine-ui` branch for all kinds of tests, before @SuperQ suggested limiting Go memory usage, which I did in https://github.com/prometheus/prometheus/pull/14862 (also included in this PR branch already). Maybe we need to be even more conservative with the memory usage and reduce it to like 8GB or something?

juliusv commented:
Yeah, it failed with this error again:

```
# github.com/prometheus/prometheus/discovery/azure.test
/usr/local/go/pkg/tool/linux_amd64/link: mapping output file failed: no space left on device
FAIL	github.com/prometheus/prometheus/discovery/azure [build failed]
```

juliusv commented:
Just to clarify, besides that CI issue: Does merging this branch mean that we're closed for any further changes in 3.0.0-beta.0? There's a few last-minute improvements/fixes in `main` that would still be nice to include (if possible), especially https://github.com/prometheus/prometheus/pull/14888

Nexucis commented:
@roidelapluie and @SuperQ are patching the CI with the artifacts. I think we will need to merge back the main branch here to have the fix and includes the fix you mentioned @juliusv makes sense to me

jan--f commented:
> Just to clarify, besides that CI issue: Does merging this branch mean that we're closed for any further changes in 3.0.0-beta.0? There's a few last-minute improvements/fixes in `main` that would still be nice to include (if possible), especially #14888

I was going to merge `release-3.0` into `main` as soon as this here lands and add the release tags there.

jan--f commented:
I can also prep another branch sync, but it would be good to merge this here first to not create a conflict for the CHANGELOG.md file.

Nexucis reviewed (approved):
So amazing !!! 🤩
